### PR TITLE
add cloudflare worker tests for analytics-node

### PR DESCRIPTION
**Pull request by chrisradek**
This PR adds cloudflare worker tests to the node-integration-tests package. This uses the latest wrangler to build and run the tests, which is using workerd under the hood. Basically as close to the real thing as we can get.

### CODE REVIEW

**Review by silesky — APPROVED** [review]

**Comment by silesky** [comment]
Looks good!  Can you update the README? https://github.com/segmentio/analytics-next/tree/master/packages/node-integration-tests/README.md

File: packages/node-integration-tests/package.json [hunk 1]
```diff
@@ -10,16 +10,20 @@
     "durability": "yarn ts-node src/durability-tests/durability-tests.ts",
     "concurrently": "yarn run -T concurrently",
     "ts-node": "yarn run -T ts-node",
-    "test": "yarn perf && yarn durability"
+    "test": "yarn perf && yarn durability",
+    "cloudflare-workers": "yarn run -T jest src/cloudflare"
```
Comment (silesky) [inline packages/node-integration-tests/package.json:14]: Nit: maybe "test:cf-workers" would be more consistent?

File: packages/node-integration-tests/src/common/mock-segment.ts [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+function isBatchRequest(req: IncomingMessage) {
+  if (req.url?.endsWith('/v1/batch')) {
+    return true
+  }
+  return false
+}
+
+type BatchHandler = (batch: any) => void
+
+export class MockSegmentServer {
```
Comment (silesky) [inline packages/node-integration-tests/src/common/mock-segment.ts:25]: You know we have this, which also is a mock segment server?  https://github.com/segmentio/analytics-next/tree/master/packages/node-integration-tests/src/server

**Comment by silesky** [comment]
Approving because not of the suggestions are blocking. This looks awesome!

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -27,6 +27,27 @@ jobs:
             ${{ runner.os }}-turbo-
       - run: yarn turbo run --filter='./packages/node*' lint
       - run: yarn turbo run --filter='./packages/node*' test
+  analytics-node-cf-workers:
+    name: 'analytics-node QA (Cloudflare Workers)'
```
Comment (silesky) [inline .github/workflows/ci.yml:31]: Might it make sense to just fold this into the consent-intg-tests job and run everything on node 18? Naming is a bit weird now with a second category of  integration tests

File: packages/node-integration-tests/package.json (hunk 1 above)
Comment (chrisradek) [reply packages/node-integration-tests/package.json:14]: Totally! Still make sense to keep cloudflare workers out of the main `test` script? Originally had them in, only reason to keep out is that they require a higher node.js runtime that the other tests do.

Comment (silesky) [reply packages/node-integration-tests/package.json:14]: Maybe we just remove the basic test script altogether?

**Comment by chrisradek** [comment]
Thanks for the feedback! Went ahead and implemented all your suggestions, merging!

**Pull request merged**